app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -231,25 +235,44 @@ export default function ManageChannelMembers({
 
     useNavButtonPressed(MANAGE_BUTTON, componentId, toggleManageEnabled, [toggleManageEnabled]);
 
-    useEffect(() => {
-        mounted.current = true;
-        const options: GetUsersOptions = {sort: 'admin', active: true, per_page: PER_PAGE_DEFAULT};
+    const handleReachedBottom = useCallback(() => {
+        if (hasMoreProfiles.current && !loading && !searchedTerm) {
+            pageRef.current += 1;
+            setLoading(true);
+            getFetchChannelMembers();
+        }
+    }, [loading, searchedTerm]);
+
+    const getFetchChannelMembers = useCallback(() => {
+        const options: GetUsersOptions = {sort: 'admin', active: true, per_page: PER_PAGE_DEFAULT, page: pageRef.current};
         fetchChannelMemberships(serverUrl, channelId, options, true).then(({users, members}) => {
             if (!mounted.current) {
                 return;
             }
-
-            if (users.length >= PER_PAGE_DEFAULT) {
-                hasMoreProfiles.current = true;
+            if (users.length < PER_PAGE_DEFAULT) {
+                hasMoreProfiles.current = false;
             }
             if (users.length) {
-                setProfiles(users);
-                setChannelMembers(members);
+                // eslint-disable-next-line max-nested-callbacks
+                setChannelMembers((prev) => [...prev, ...members]);
+                // eslint-disable-next-line max-nested-callbacks
+                setProfiles((prev) => [...prev, ...users]);
             }
+
             setLoading(false);
         });
+    }, []);
+
+    useEffect(() => {
+        mounted.current = true;
+        pageRef.current = 0;
+        hasMoreProfiles.current = true;
+
+        getFetchChannelMembers();
         return () => {
+            hasMoreProfiles.current = false;
             mounted.current = false;
+            pageRef.current = 0;
```

Comment:
Not sure why we set these on unmount.

The reason we set `mounted` is because async callbacks may finish after the component has been destroyed, so we want to avoid useless operations, and potentially setting the state on a destroyed component.

But not sure why we would want to reset the page or the `hasMoreProfiles`.